pennylane/ops/qutrit/parametric_ops.py

```diff
@@ -0,0 +1,173 @@
+
+    def __init__(
+        self, phi, wires, subspace=[0, 1], do_queue=True, id=None
+    ):  # pylint: disable=dangerous-default-value
+        self._subspace = self.validate_subspace(subspace)
+        self._hyperparameters = {
+            "subspace": self._subspace,
+        }
+        super().__init__(phi, wires=wires, do_queue=do_queue, id=id)
+
+    @property
+    def subspace(self):
+        """The single-qutrit basis states the operator acts on.
+
+        The subspace defines which two basis states the opreation acts on. The basis state
+        not included in the subspace remains unaffected.
+
+        Returns:
+            tuple[int]: subspace on which the operator acts
+        """
+        return self._subspace
+
+    @staticmethod
+    def compute_matrix(
+        theta, subspace=(0, 1)
+    ):  # pylint: disable=arguments-differ,dangerous-default-value
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.TRX.matrix`
+
+        Args:
+            theta (tensor_like or float): rotation angle
+            subspace (Sequence[int]): the 2D subspace on which to apply the operation
+
+        Returns:
+            tensor_like: canonical matrix
+
+        **Example**
+
+        >>> qml.TRX.compute_matrix(torch.tensor(0.5), subspace=[0, 2])
+        tensor([[0.9689+0.0000j, 0.0000+0.0000j, 0.0000-0.2474j],
+                [0.0000+0.0000j, 1.0000+0.0000j, 0.0000+0.0000j],
+                [0.0000-0.2474j, 0.0000+0.0000j, 0.9689+0.0000j]])
+        """
+        c = qml.math.cos(theta / 2)
+        s = qml.math.sin(theta / 2)
+
+        if qml.math.get_interface(theta) == "tensorflow":
+            c = qml.math.cast_like(c, 1j)
+            s = qml.math.cast_like(s, 1j)
+
+        # The following avoids casting an imaginary quantity to reals when backpropagating
+        c = (1 + 0j) * c
+        js = -1j * s
+
+        shape = qml.math.shape(theta)
+        is_broadcasted = len(shape) != 0 and shape[0] > 1
+        # Construct identity matrices and cast to complex type
+        mat = (
+            qml.math.tensordot([1] * qml.math.shape(theta)[0], qml.math.eye(3), axes=0)
+            if is_broadcasted
+            else qml.math.eye(3)
+        )
+        mat = qml.math.cast_like(mat, js)
+        # Create slices that determine the indices at which the rotation terms of the matrix should be
+        slices = tuple(itertools.product(subspace, subspace))
+        if is_broadcasted:
+            slices = [(Ellipsis, *s) for s in slices]
+
+        # Put rotation terms in the appropriate indices using the slices
+        mat[slices[0]] = mat[slices[3]] = c
+        mat[slices[1]] = mat[slices[2]] = js
+        return qml.math.convert_like(mat, theta)
+
```

Comment:
Do we need `compute_eigvals` as well?

Reply:
Okay - maybe good to get a second opinion from someone else on the team? (I think we need a second reviewer anyways still to approve)